knowhere/index/vector_index/helpers/IndexParameter.h

```diff
@@ -133,32 +162,35 @@ DEFINE_CONFIG_SETTER(SetMetaTraceVisit, meta::TRACE_VISIT, bool)
 
 ///////////////////////////////////////////////////////////////////////////////
 // APIs to access indexparam
-
-DEFINE_CONFIG_GETTER(GetIndexParamNprobe, indexparam::NPROBE, int64_t)
+static const int64_t DEFAULT_NPROBE = 8;
+static const int64_t DEFAULT_NLIST = 128;
+static const int64_t DEFAULT_PQ_M = 4;
+static const int64_t DEFAULT_PQ_NBITS = 8;
+static const int64_t DEFAULT_HNSW_EFCONSTRUCTION = 360;
+static const int64_t DEFAULT_HNSW_M = 30;
+
+DEFINE_CONFIG_GETTER_WITH_DEFAULT_VALUE(GetIndexParamNprobe, indexparam::NPROBE, DEFAULT_NPROBE, int64_t)
 DEFINE_CONFIG_SETTER(SetIndexParamNprobe, indexparam::NPROBE, int64_t)
 
-DEFINE_CONFIG_GETTER(GetIndexParamNlist, indexparam::NLIST, int64_t)
+DEFINE_CONFIG_GETTER_WITH_DEFAULT_VALUE(GetIndexParamNlist, indexparam::NLIST, DEFAULT_NLIST, int64_t)
 DEFINE_CONFIG_SETTER(SetIndexParamNlist, indexparam::NLIST, int64_t)
 
-DEFINE_CONFIG_GETTER(GetIndexParamNbits, indexparam::NBITS, int64_t)
+DEFINE_CONFIG_GETTER_WITH_DEFAULT_VALUE(GetIndexParamNbits, indexparam::NBITS, DEFAULT_PQ_NBITS, int64_t)
 DEFINE_CONFIG_SETTER(SetIndexParamNbits, indexparam::NBITS, int64_t)
 
 // PQ param for IVFPQ
-DEFINE_CONFIG_GETTER(GetIndexParamM, indexparam::M, int64_t)
+DEFINE_CONFIG_GETTER_WITH_DEFAULT_VALUE(GetIndexParamM, indexparam::M, DEFAULT_PQ_M, int64_t)
 DEFINE_CONFIG_SETTER(SetIndexParamM, indexparam::M, int64_t)
 
-// PQ param for RHNSWPQ
-DEFINE_CONFIG_GETTER(GetIndexParamPQM, indexparam::PQ_M, int64_t)
-DEFINE_CONFIG_SETTER(SetIndexParamPQM, indexparam::PQ_M, int64_t)
-
 // HNSW Params
-DEFINE_CONFIG_GETTER(GetIndexParamEfConstruction, indexparam::EFCONSTRUCTION, int64_t)
+DEFINE_CONFIG_GETTER_WITH_DEFAULT_VALUE(GetIndexParamEfConstruction, indexparam::EFCONSTRUCTION,
+                                        DEFAULT_HNSW_EFCONSTRUCTION, int64_t)
 DEFINE_CONFIG_SETTER(SetIndexParamEfConstruction, indexparam::EFCONSTRUCTION, int64_t)
 
-DEFINE_CONFIG_GETTER(GetIndexParamHNSWM, indexparam::HNSW_M, int64_t)
+DEFINE_CONFIG_GETTER_WITH_DEFAULT_VALUE(GetIndexParamHNSWM, indexparam::HNSW_M, DEFAULT_HNSW_M, int64_t)
 DEFINE_CONFIG_SETTER(SetIndexParamHNSWM, indexparam::HNSW_M, int64_t)
 
-DEFINE_CONFIG_GETTER(GetIndexParamEf, indexparam::EF, int64_t)
+DEFINE_CONFIG_GETTER_WITH_ALTER_KEY(GetIndexParamEf, indexparam::EF, meta::TOPK, int64_t)
```

Comment:
ef = k < 16 ? 16:k;
The third parameters in DEFINE_CONFIG_GETTER_WITH_ALTER_KEY  should be a T type value, max(16, k), instead of  a string key.

Reply:
as GaoChao's document, 'ef' 's default value is 'topk'.
![Screenshot from 2023-05-28 21-53-00](https://github.com/milvus-io/knowhere/assets/3992404/8f788555-3bb4-4a44-bb38-1995cabad45a)